proknow/Audit.py

```diff
@@ -0,0 +1,169 @@
+                    options["status_codes"] = kwargs["status_codes"]
+
+            if "text" in kwargs: 
+                if not isinstance(kwargs["text"], list):
+                    options["text"] = [kwargs["text"]]
+                else:
+                    options["text"] = kwargs["text"]
+
+        return self._query(options)
+
+    def _query(self, options): 
+        _, results = self._requestor.post('/audit/events/search', json=options)
+        
+        return AuditResultsPage(self, options, results['total'], results['items'])
+
+class AuditResultsPage(object):
+    """
+    This class represents a page of audit log query results. It's instantiated by the
+    :meth:`proknow.Audit.Audit.query` method.
+
+    Attributes:
+        total (int): The total number of possible results for the given query, not the total of items in the object's item list.
+        items (list of dict): A list of dictionaries to represent each item returned for the current page.
+
+            Item Keys:
+                * id: The ID of the event
+                * timestamp: Time of event
+                * user_id: User ID of event's enacting user
+                * user_name: User name of event's enacting user
+                * patient_id: ID of patient
+                * patient_name: Name of patient
+                * patient_mrn: Medical record number of patient
+                * resource_id: ID of a resource
+                * resource_name: Name of a resource
+                * workspace_id: ID of workspace in which events took place
+                * workspace_name: Name of workspace in which events took place
+                * collection_id: ID of a collection
+                * type: Event type (e.g. 'workspace_created')
+                * classification: 'HTTP' or 'AUTH'
+                * method: HTTP method: 'GET', 'HEAD', 'POST', 'PUT', 'DELETE', 'CONNECT', 'OPTIONS', 'TRACE', or 'PATCH'
+                * uri: Uri that event took place (e.g. '/metrics/custom')
+                * user_agent: User Agent attributed to event (e.g. Browser User Agent)
+                * ip_address: IP Address attributed to event
+                * status_code: Status code of event (e.g. 200)
+                * data: Request data
+    """
+
+    def __init__(self, audit, options, total, items):
+        self._audit = audit
+        self._options = options
+        self.total = total
+        self.items = items
+        if "page_number" not in self._options:
+            self._options["page_number"] = 0
+        if "first_id" not in self._options and total > 0:
+            self._options["first_id"] = items[0]["id"]
+    
+    def next(self):
+        """Gets the next page of query results using the initial query parameters.
+
+        Returns:
+            :class:`proknow.Audit.AuditResultsPage`
+
+        Raises: 
+            :class:`proknow.Exceptions.HttpError`: If the HTTP request generated an error.
+        """
+        self._options["page_number"] += 1
```

Comment:
Rather than mutate the options object, I think this should clone the dictionary, increment the page number on the clone, and pass the clone to query method. Take a look at this example:

```python
results1 = pk.audit.query()
results2 = results1.next()
results3 = results1.next()
results4 = results2.next()
results5 = results3.next()
```

With the way it's currently implemented, `results1` would be page 1, `results2` would be page 2, `results3` would be page 3, and so on. However, I think what I'd expect is for `results2` and `results3` to be page 2 and `results4` and `results5` to be page 3.